Tests/Editor/Tracking/Collision/List/SortCollisionsByNearestTest.cs

```diff
@@ -0,0 +1,89 @@
+    using Test.VRTK.Core.Utility.Mock;
+    using System.Collections.Generic;
+
+    public class SortCollisionsByNearestTest
+    {
+        private GameObject containingObject;
+        private SortCollisionsByNearest subject;
+
+        [SetUp]
+        public void SetUp()
+        {
+            containingObject = new GameObject();
+            subject = containingObject.AddComponent<SortCollisionsByNearest>();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            Object.DestroyImmediate(subject);
+            Object.DestroyImmediate(containingObject);
+        }
+
+        [Test]
+        public void Sort()
+        {
+            UnityEventListenerMock sortedMock = new UnityEventListenerMock();
+            subject.Sorted.AddListener(sortedMock.Listen);
+
+            List<CollisionNotifier.EventData> collisionList = new List<CollisionNotifier.EventData>();
+            GameObject oneContainer;
+            CollisionNotifier.EventData oneData = GenerateCollision(out oneContainer, Vector3.one * 5f);
+            oneContainer.name = "one";
+            collisionList.Add(oneData);
+
+            GameObject twoContainer;
+            CollisionNotifier.EventData twoData = GenerateCollision(out twoContainer, Vector3.one * 2f);
+            twoContainer.name = "two";
+            collisionList.Add(twoData);
+
+            GameObject threeContainer;
+            CollisionNotifier.EventData threeData = GenerateCollision(out threeContainer, Vector3.one * 3f);
+            threeContainer.name = "three";
+            collisionList.Add(threeData);
+
+            ActiveCollisions.EventData eventData = new ActiveCollisions.EventData().Set(collisionList);
+
+            subject.source = containingObject.transform;
+
+            Assert.AreEqual("one,two,three,", GetListNames(eventData));
+            Assert.IsFalse(sortedMock.Received);
+
+            ActiveCollisions.EventData sortedList = subject.Sort(eventData);
+
+            Assert.IsTrue(sortedMock.Received);
+            Assert.AreEqual("two,three,one,", GetListNames(sortedList));
+
+            Object.DestroyImmediate(oneContainer);
+            Object.DestroyImmediate(twoContainer);
+            Object.DestroyImmediate(threeContainer);
+        }
+
+        protected string GetListNames(ActiveCollisions.EventData list)
+        {
+            string returnString = "";
+            foreach (CollisionNotifier.EventData data in list.activeCollisions)
+            {
+                returnString += data.collider.name + ",";
+            }
+            return returnString;
+        }
+
+        protected CollisionNotifier.EventData GenerateCollision(out GameObject container, Vector3 position)
+        {
+            container = new GameObject();
+            BoxCollider boxCollider = container.AddComponent<BoxCollider>();
+            Collider collider = container.GetComponent<Collider>();
+            collider.transform.position = position;

```

Comment:
Despite this being different to the two other copies of this method we could still merge them into a method used in all these three places.